onnxruntime/core/providers/cpu/nn/batch_norm_helper.h

```diff
@@ -14,48 +14,92 @@ class BatchNormHelper {
                                        const Tensor* scale,
                                        const Tensor* B,
                                        const Tensor* mean,
-                                       const Tensor* var) {
+                                       const Tensor* var,
+                                       bool is_spatial = true) {
+    const auto& x_dims = X->Shape().GetDims();
+    if (x_dims.size() < 2) {
```

Comment:
Even though, the spec doesn't explicitly state it, it makes sense if the minimum rank of 'X' is atleast 3 - N,C, and atleast one data dimension (D1). However, when we impose that, it seems to break some successfully running models that have input rank 2 - [N, C]. The implicit shape is [N, C, 1] in those cases and the results seem ok. Hence, modifying the minimum rank criteria to just 2. This is okay to add as in the next step we parse the number of channels in the index position 1.